tests/test_reply_headers.js

```diff
@@ -364,22 +363,20 @@ describe('`replyDate()`', () => {
       }
     })
 
-    // async / got version is returning "not ok test unfinished".
-    // https://github.com/nock/nock/issues/1305#issuecomment-451701657
-    it('sends date header with response', done => {
+    it('sends date header with response', async () => {
       const date = new Date()
 
       const scope = nock('http://example.test')
         .replyDate()
         .get('/')
         .reply()
 
-      mikealRequest.get('http://example.test', (err, resp) => {
-        expect(err).to.be.null()
-        expect(resp.headers.date).to.equal(date.toUTCString())
-        scope.done()
-        done()
-      })
+      const req = got('http://example.test/')
+      clock.tick()
```

Comment:
`clock.tick()` was the key to making this work.

Reply:
It's unclear to me at a glance if creating the `date` var before the `clock.tick()` is important. 
If it's not, could we move it down to just above the assertion that uses it?

Reply:
It would be better if it were after the `clock.tick()` and I've moved it. I'm surprised that it can be just before the assertion and still work, though it seems to! Happy to keep it there. I don't feel like this mock timers business is something I need to fully grasp.